extensions/amp-sidebar/0.1/swipe-to-dismiss.js

```diff
@@ -0,0 +1,355 @@
+   * @private
+   */
+  adjustForSwipePosition_(swipeElementTransform = '', maskOpacity = '') {
+    setStyles(dev().assertElement(this.swipeElement_), {
+      transform: swipeElementTransform,
+      transition: '',
+    });
+    setStyles(dev().assertElement(this.mask_), {
+      opacity: maskOpacity,
+      transition: '',
+    });
+  }
+
+  /**
+   * Releases the user's swipe to dismiss gesture. This carries the momentum
+   * forwards and either closes the lightbox or snaps back based on the speed
+   * and distance. This should be called in a mutate context.
+   * @param {number} velocityX The X velocity when the swipe was released.
+   * @param {number} velocityY The Y velocity when the swipe was released.
+   * @param {number} deltaX The x distance when the swipe was released.
+   * @param {number} deltaY The y distance when the swipe was released.
+   * @return {!Promise} A Promise that resolves once the release is completed,
+   *    either snapping back to the start or closing the carousel.
+   * @private
+   */
+  releaseSwipe_(velocityX, velocityY, deltaX, deltaY) {
+    const distanceX = velocityX * SWIPE_TO_CLOSE_VELOCITY_TO_DISTANCE_FACTOR;
+    const distanceY = velocityY * SWIPE_TO_CLOSE_VELOCITY_TO_DISTANCE_FACTOR;
+    const finalDeltaX = distanceX + deltaX;
+    const finalDeltaY = distanceY + deltaY;
+    // We want to figure out the final distance we will rest at if the user
+    // flicked the element and use that to determine we should animate to. We
+    // will then use that resting position to determine if we should snap back
+    // or close.
+    const finalDistance = this.capDistance_(finalDeltaX, finalDeltaY);
+    const finalVelocity = this.capDistance_(velocityX, velocityY);
+
+    const swipeToCloseDistance =
+      this.getSwipeElementLength_() * SWIPE_TO_CLOSE_DISTANCE_THRESHOLD;
+    if (
+      finalDistance < swipeToCloseDistance &&
+      finalVelocity < SWIPE_TO_CLOSE_VELOCITY_THRESHOLD
+    ) {
+      return this.carrySwipeMomentum_(finalDistance, finalVelocity).then(() =>
+        this.snapBackFromSwipe_(finalDistance)
+      );
+    }
+
+    const remainingDistance = this.getSwipeElementLength_() - finalDistance;
+    const duration = remainingDistance * SWIPE_TO_CLOSE_DISTANCE_TO_TIME_FACTOR;
+
+    return this.mutateElement_(() => {
+      setStyles(dev().assertElement(this.swipeElement_), {
+        transform: this.translateBy_(100, '%'),
+        transition: `${duration}ms transform ease-out`,
+      });
+      setStyles(dev().assertElement(this.mask_), {
+        opacity: 0,
+        transition: `${duration}ms opacity ease-out`,
+      });
+    })
+      .then(() => delayAfterDeferringToEventLoop(this.win_, duration))
+      .then(() => this.onclose_());
+  }
+
+  /**
+   * Handles the start of a swipe to dimiss gesture:
+   *  - Prevents a scroll event from the element during the swipe.
+   * This should be called in a mutate context.
+   * @private
+   */
+  startSwipeToDismiss_() {
```

Comment:
Lets remove this method, while it was useful for the lightbox swipe to dismiss, I don't think it is needed for this one.

Reply:
I'll remove the `endSwipeToDismiss` and `preventScrollUnlistener` as well